winml/dll/module.cpp

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include <windows.h>
+#include <Hstring.h>
+
+#include "LearningModelDevice.h"
+#include "OnnxruntimeProvider.h"
+
+using namespace winrt::Windows::AI::MachineLearning::implementation;
+
+void __stdcall OnErrorReported(bool alreadyReported, wil::FailureInfo const& failure) WI_NOEXCEPT {
+  if (!alreadyReported) {
+    winrt::hstring message(failure.pszMessage ? failure.pszMessage : L"");
+    telemetry_helper.LogRuntimeError(
+        failure.hr,
+        winrt::to_string(message),
+        failure.pszFile,
+        failure.pszFunction,
+        failure.uLineNumber);
+  }
+}
+
+extern "C" BOOL WINAPI DllMain(_In_ HINSTANCE hInstance, DWORD dwReason, _In_ void* lpvReserved) {
+  switch (dwReason) {
+    case DLL_PROCESS_ATTACH:
+      DisableThreadLibraryCalls(hInstance);
```

Comment:
"DisableThreadLibraryCalls(hInstance)"
Is WinML magic statics free? If you have any magic static variables, this doesn't work.

Reply:
You are correct. We noticed and did a major statics cleanup a little while ago. there are still some exceptions, but we will resolve those statics soon.